package-parser/package_parser/commands/get_api/_model.py

```diff
@@ -350,31 +350,90 @@ def to_json(self) -> Any:
         }
 
 
-class RefinedType:
+class Type:
+    def __init__(
+        self,
+        typestring: ParameterAndResultDocstring,
+    ) -> None:
+        self.type: Union[
+            NamedType, EnumType, BoundaryType, UnionType, None
+        ] = Type.create_type(typestring)
+
     @classmethod
-    def from_docstring(cls, docstring: ParameterAndResultDocstring) -> RefinedType:
-        docstring_str = " ".join([docstring.type, docstring.description])
-        enum = EnumType.from_string(docstring_str)
-        boundary = BoundaryType.from_string(docstring_str)
+    def create_type(
+        cls, docstring: ParameterAndResultDocstring
+    ) -> Union[NamedType, EnumType, BoundaryType, UnionType, None]:
+        type_string = docstring.type
+        types: list[Union[NamedType, EnumType, BoundaryType]] = list()
 
-        if enum is not None:
-            return RefinedType(enum)
+        # Collapse whitespaces
+        type_string = re.sub(r"\s+", " ", type_string)
 
+        # Get boundary from description
+        boundary = BoundaryType.from_string(docstring.description)
```

Comment:
Is there a reason why BoundaryType gets the entire description as input, when we use `docstring.type` for everything else?